packages/openapi-generator/src/parser/document.ts

```diff
@@ -28,7 +26,7 @@ export async function parseOpenApiDocument(
 
   return {
     apis: parseApis(document, refs, options),
-    serviceName: pascalCase(serviceName),
+    serviceName: pascalCase(serviceOptions.serviceName),
```

Comment:
I think the service name in the service options should already be pascalCase, don't you think? If users use a non-pascal case name here I would say that's ok.